From 6ded2655dbec92751cfb14b8b4ddaeaf89a66a6e Mon Sep 17 00:00:00 2001
From: Rui Wang <ruwang@redhat.com>
Date: Fri, 22 Aug 2014 03:52:08 -0400
Subject: [acpi] x86/ras: Fix build warnings in <linux/aer.h>

Message-id: <1408679528-32070-11-git-send-email-ruwang@redhat.com>
Patchwork-id: 88699
O-Subject: [PATCH RHEL7.1 BZ1085141 10/10] x86/ras: Fix build warnings in <linux/aer.h>
Bugzilla: 1085141
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: "Chen, Gong" <gong.chen@linux.intel.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1085141

Backport of:
Original commit: 5ccb8225abf2ac51cd023a99f28366ac9823bd0d
Original Author: Mike Qiu <qiudayu@linux.vnet.ibm.com>
Original Date: Tue, 29 Jul 2014 10:49:25 -0700

 Fix build warning due to a missing forward declaration in
 <linux/aer.h>.  We need struct pci_dev to be forward declared so we
 can define pointers to it, but we don't need to pull in the whole
 definition.

 build log:

 In file included from include/ras/ras_event.h:11:0,
    from drivers/ras/ras.c:13:
 include/linux/aer.h:42:129: warning: struct pci_dev
 declared inside parameter list [enabled by default]

 include/linux/aer.h:42:129: warning: its scope is only
 this definition or declaration, which is probably not
 what you want [enabled by default]

 include/linux/aer.h:46:130: warning: struct pci_dev
 declared inside parameter list [enabled by default]

 include/linux/aer.h:50:136: warning: struct pci_dev
 declared inside parameter list [enabled by default]

 include/linux/aer.h:57:14: warning: struct pci_dev
 declared inside parameter list [enabled by default]

Signed-off-by: Chen, Gong <gong.chen@linux.intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/aer.h b/include/linux/aer.h
index 4dbaa70..c826d1c 100644
--- a/include/linux/aer.h
+++ b/include/linux/aer.h
@@ -11,6 +11,8 @@
 #define AER_FATAL   1
 #define AER_CORRECTABLE   2
 
+struct pci_dev;
+
 struct aer_header_log_regs {
  unsigned int dw0;
  unsigned int dw1;
-- 
1.7.1